gcp/cloud-run-v2/main.tf

```diff
@@ -197,30 +209,68 @@ module "lb-http" {
   https_redirect            = true # Enable HTTPS redirect
   random_certificate_suffix = true
 
-  backends = {
-    default = {
-      groups = [
-        {
-          group = google_compute_region_network_endpoint_group.cloudrun_neg[count.index].id
-        }
-      ]
+  url_map        = var.create_url_map == false? google_compute_url_map.custom_url_map_https[count.index].self_link : null
+  create_url_map = var.create_url_map
 
-      description             = "Backend for Cloud Run service"
-      enable_cdn              = false
-      custom_request_headers  = ["X-Client-Geo-Location: {client_region_subdivision}, {client_city}"]
-      custom_response_headers = ["X-Cache-Hit: {cdn_cache_status}"]
+  backends = merge(
+    {
+      "default" = merge(local.default_backend_config, {
+        groups = [
+          {
+            group = google_compute_region_network_endpoint_group.cloudrun_neg[0].id
+          }
+        ]
+        security_policy = var.cloud_armor.enabled ? google_compute_security_policy.cloud_armor_policy[0].self_link : null
+      })
+    },
+    { for key, value in var.additional_backend_services :
+      key => merge(local.default_backend_config, {
+        groups = [
+          {
+            group = value.group
+          }
+        ]
+        security_policy = value.cloud_armor ? google_compute_security_policy.cloud_armor_policy[0].self_link : null
+      })
+    }
+  )
+}
 
-      # Clour Armor security
-      security_policy = var.cloud_armor.enabled ? google_compute_security_policy.cloud_armor_policy[0].self_link : null
+resource "google_compute_url_map" "custom_url_map_https" {
+  count           = var.create_url_map == false ? 1 : 0
+  name            = "${var.name}-https-urlmap"
+  description     = "Custom URL map for Cloud Run service"
+  default_service = module.lb-http[0].backend_services["default"].self_link
 
-      log_config = {
-        enable = false
-      }
-      iap_config = {
-        enable = false
+  host_rule {
+    hosts        = [local.domain]
+    path_matcher = "allpaths"
+  }
+  path_matcher {
+    name            = "allpaths"
+    default_service = module.lb-http[0].backend_services["default"].self_link
+    dynamic "path_rule" {
+      for_each = var.path_rules
+      content {
+        paths   = length(path_rule.value.paths)  > 0 ? path_rule.value.paths : ["/*"]
+        service = path_rule.value.service_name != null ? module.lb-http[0].backend_services["${path_rule.value.service_name}"].self_link : module.lb-http[0].backend_services["default"].self_link
+        dynamic "route_action" {
+          for_each = path_rule.value.route_action != null ? [1] : []
+          content {
+            url_rewrite {
+              path_prefix_rewrite = path_rule.value.route_action.url_rewrite.path_prefix_rewrite != null ? path_rule.value.route_action.url_rewrite.path_prefix_rewrite : null
+            }
+          }
+          
+        }
       }
     }
   }
+  test {
```

Comment:
What is this

Reply:
it's for testing proposes but don't think is necessary so I am removing this